code/__HELPERS/game.dm

```diff
@@ -159,6 +159,37 @@
 		processing_list.Cut(1, 2)
 		processing_list += A.contents
 
+// Recursive loop based off of the recursive mob check. Necessary for preventing organ decay without running every tick.
+// Takes in a variable "atom/O" and looks for any/all organs to toggle decompose, excluding other organ safe containers
+/proc/recursive_organ_check(atom/O, var/decompose)
+
+	var/list/processing_list = list(O)
+	var/list/processed_list = list()
+	var/obj/item/organ/found_organ
+
+	while(processing_list.len)
+
+		var/atom/A = processing_list[1]
+
+		if(istype(A, /obj/item/organ))
+			found_organ = A
+			found_organ.can_decompose = decompose
+
+		else if(istype(A, /mob/living/carbon))
+			var/mob/living/carbon/Q = A
+			for(var/organ in Q.internal_organs)
+				found_organ = organ
+				found_organ.can_decompose = decompose
+
+		for(var/atom/B in A)
+			if(!processed_list[B] && !istype(B, /obj/structure/closet/crate/freezer) && !istype(B, /obj/structure/closet/secure_closet/freezer))
+				processing_list|= B
+
+		processing_list.Cut(1, 2)

```

Comment:
instead of doing this cut and always pulling from 1, keep track of a working index.

Reply:
Since this list is storing variables by object, and not by number, using a working index will be difficult with what I know of byond lists, which is limited. Trying an index++ and replacing the if condition with whether index > list.len doesn't work, since the objects are stored as processing_list[organ] not processing_list[1] afaik, but I'll look into it more later.

Reply:
lists are both at the same time when associated, you can reference them by index or by the object being stored. It's still messy but check out propagate_network() in power.dm#212 for a working example of the structure of the loop.

Reply:
Not sure what changed, used index < length(list) for the while condition and it works now. Thanks for the reference to an example in code btw!